src/Illuminate/Auth/SessionGuard.php

```diff
@@ -419,13 +430,17 @@ public function attemptWhen(array $credentials = [], $callbacks = null, $remembe
      */
     protected function hasValidCredentials($user, $credentials)
     {
-        $validated = ! is_null($user) && $this->provider->validateCredentials($user, $credentials);
+        return $this->timebox->make(function ($timebox) use ($user, $credentials) {
+            $validated = ! is_null($user) && $this->provider->validateCredentials($user, $credentials);
 
-        if ($validated) {
-            $this->fireValidatedEvent($user);
-        }
+            if ($validated) {
+                $timebox->returnEarly();
+
+                $this->fireValidatedEvent($user);
+            }
 
-        return $validated;
+            return $validated;
+        }, 200000);
```

Comment:
what do you think about making this magic number configurable?

Reply:
It should be configurable, because some services (like AWS) will bill you for the time.

Also there is a concern of DoS because your service will block for a certain amount of time, which means some connections like cache, database and etc. will be kept open for the same amount of time which will increase the amount of concurrent connections open. (Some services have limitations for the amount of concurrent connections open)

Reply:
Good idea. Could it be `auth.validate_credentials_minimum_time` or do you have better suggestions for an config key?

Reply:
I have placed the config inside the auth.guards.session.validate_credentials_minimum_time now.
The SessionGuard operates in that scope so thought it made sense to place it there. The default will still be 200000 microseconds.